### PR TITLE
Bug 1167595: Support template strings.

### PR DESCRIPTION
**Pull request by kmaglione**

### CODE REVIEW

File: tests/test_js_syntax.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from nose.tools import eq_
 
-from js_helper import _do_test_raw
+from .js_helper import TestCase, _do_test_raw
```
Comment (magopian) [inline tests/test_js_syntax.py]: Is it a letter before an underscore in the "alphabetical order" we use for imports?

File: jetpack/addon-sdk [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 74d7ae8f5496979baf1b7e995b38474dc81d5e70
+Subproject commit e2b154aff5c452d3b4f25bd67513122bd6172319
```
Comment (magopian) [inline jetpack/addon-sdk]: is that changed on purpose?

File: validator/testcases/javascript/actions.py [hunk 3]
```diff
@@ -799,34 +818,33 @@ def _expr_binary(traverser, node):
...
-        # of binary expressions.
-        left = _expr_binary(traverser, node["left"])
-        node["left"]["__traversal"] = left
-    else:
-        left = traverser._traverse_node(node["left"])
+    with traverser._debug("BIN_EXP>>l-value"):
+        if (node["left"]["type"] == "BinaryExpression" and
+            "__traversal" not in node["left"]):
+            # Process the left branch of the binary expression directly. This
+            # keeps the recursion cap in line and speeds up # processing of
```
Comment (magopian) [inline validator/testcases/javascript/actions.py]: there's a rogue `#` in there

**Comment by magopian** [comment]
Looks nice, but there's modifications that are unrelated to the bug, right?

r+wc

File: tests/test_js_syntax.py (hunk 1 above)
Comment (kmaglione) [inline tests/test_js_syntax.py]: `<capital letter>` < `_` < `<lowercase letter>`, at least in ASCII sort order.

File: jetpack/addon-sdk (hunk 2 above)
Comment (kmaglione) [inline jetpack/addon-sdk]: Bah. Nope.

**Comment by kmaglione** [comment]
The subproject change is unrelated to this bug. I made the change to `traverser._debug` when I was trying to understand a failing testcase, so in that sense it's related.

Everything else, except for the removal of an unused variable, is directly related.

**Comment by kmaglione** [comment]
Updated per comments.

**Comment by magopian** [comment]
Thanks!

**Pull request merged**